docs/source/contributing.rst

```diff
@@ -200,26 +200,37 @@ Function and object doc strings
 For inline strings documenting functions and objects ("doc strings"), we use the
 `numpydoc style (Numpy docstring format) <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
 
-Sphinx ReadTheDocs documentation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Jupter Book ReadTheDocs documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Echopype documentation (`<https://echopype.readthedocs.io>`_) is based on `Jupyter Book <https://jupyterbook.org/en/stable/intro.html>`_
+which are rendered under the hood with `Sphinx <https://www.sphinx-doc.org>`_. The documentation is hosted at
+`Read The Docs <https://readthedocs.org>`_. The documentation package dependencies are found
+in the ``docs/requirements.txt`` file, and the source documentation files are in the ``docs/source`` directory. The echopype development
+conda environment will install all required dependencies.
+
+Our documentation are currently a mixture of the following file formats:
 
-Echopype documentation (`<https://echopype.readthedocs.io>`_) is based on
-`Sphinx <https://www.sphinx-doc.org>`_ and is hosted at
-`Read The Docs <https://readthedocs.org>`_. The sphinx files are found
-in the ``docs`` directory, and the source documentation files, written in
-`reStructuredText <https://www.sphinx-doc.org/en/main/usage/restructuredtext/index.html>`_
-(``.rst``) format, are in the ``docs/source`` directory. The echopype development
-conda environment will install all required Sphinx dependencies.
-To run Sphinx locally:
+- `CommonMark <https://commonmark.org/>`_ and `MySt <https://jupyterbook.org/en/stable/content/myst.html>`_ Markdown
+- `Jupyter Notebook <https://jupyter-notebook.readthedocs.io/en/latest/notebook.html>`_
+- `reStructuredText <https://docutils.sourceforge.io/rst.html>`_
+
+To run Jupyter Book locally:
 
 .. code-block:: bash
 
-    cd docs
-    sphinx-build -b html -d _build/doctrees source _build/html
+    jupyter-book build docs/source --path-output docs
 
-To view the generated HTML files generated by Sphinx, open the
+To view the generated HTML files generated by Jupyter Book, open the
 ``docs/_build/html/index.html`` in your browser.
 
+Jupyter Book `configurations <https://jupyterbook.org/en/stable/customize/config.html>`_ can be found in the `docs/source/_config.yml` file.
+The `table of contents <https://jupyterbook.org/en/stable/structure/toc.html>`_ arragements for the sidebar can be found in ``docs/source/_toc.yml`` file.
+
+When ready to commit your changes, please run the pre-commit as that will run spell check,
+and conversion to Sphix site for ReadTheDocs.
+See `Jupyter Book <https://jupyterbook.org/en/stable/sphinx/index.html#convert-your-jupyter-book-into-a-sphinx-website>`_ for more details on this.
```

Comment:
I know that this is good practice, but I wonder whether it is better to mention this just as optional for contributors, because we do have pre-commit bot in place to fix small things. It may be too overwhelming for people who are not already familiar with these tools when clicking into this page.

Reply:
That is true. It is another thing to keep up with. I might just change it to go check the PR in case there are any `pre-commit` CI changes, in which case, they'd have to pull changes locally if they want to make further changes.

Reply:
I agree with @leewujung . Mentioning and encouraging pre-commit and having instructions for running it locally are all great. But let's not present it as a barrier or requirement.

Reply:
Changing it to go check the PR sounds good. How about something like the following?

```suggestion
When ready to commit your changes, please pull request your changes to the `stable` branch. Once the PR is submitted, the `pre-commit` CI will run for basic spelling and formatting check. Any changes from the `pre-commit` check have to be pulled to your branch (via `git pull`) before your push further commits. You will also be able to view the new built doc in the PR via the "docs/readthedocs.org:echopype" entry shown below. 

![Screen Shot 2022-04-27 at 4 16 41 PM](https://user-images.githubusercontent.com/15334215/165646718-ebfd4041-b110-4b54-a5b9-54a7a08bc982.png)
```

Reply:
That looks much better! But, I wanted to remind us that there's actually a section on [pre-commit hooks](https://echopype.readthedocs.io/en/latest/contributing.html#pre-commit-hooks) under "Tests and test infrastructure". Some of the text you've written here specifically about pre-commit could be folded into that section, especially this:
> Once the PR is submitted, the `pre-commit` CI will run for basic spelling and formatting check. Any changes from the `pre-commit` check have to be pulled to your branch (via `git pull`) before your push further commits.

Then, we can point to that section from here.

Reply:
Sounds good incorporate that. I'll let @lsetiawan decide how he wants to connect these few things. :)